### ISSUE TITLE
Can't press and hold a Pressable on Android web

### ISSUE THREAD

**Issue by elliotwaite** [necolas/react-native-web#2349]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the issue

On Android web, when you press and hold a Pressable, it automatically calls its onPressOut callback after only a short delay, even if the Pressable is still being pressed, making it impossible to press and hold a Pressable for an extended amount of time.

### Expected behavior

I expected the Pressable to call its onPressOut callback only after it is no longer being pressed.

### Steps to reproduce

1. Load this Snack in a browser on an Android phone (I tested it using both Chrome and Firefox):
https://snack.expo.dev/@elliotwaite/press-and-hold-test
2. Load the desktop version of the site if needed to be able to interact with the web preview.
3. Try to press and hold the button.

### Test case

https://snack.expo.dev/@elliotwaite/press-and-hold-test

### Additional comments

Here's an animated gif of a recording of what happens:
![press-and-hold-demo](https://user-images.githubusercontent.com/1767836/180294813-1e9544ec-3202-4768-bafe-990b293feb86.gif)

I wasn't sure if this was an issue with react-native-web or something else. So let me know if there would be a better place for me to open an issue about this.

**Comment by necolas** [comment]
I prefer test cases use the provided codesandbox template.

I think what's happening is that you are long-touching and that triggers a `contextmenu` event which ends the press. I don't know why the contextmenu is not being displayed though, that might be a bug. If you want to own the long-press behavior, set the `onLongPress` handler.

https://codesandbox.io/s/clever-herschel-2z09xj?file=/src/App.js

**Comment by elliotwaite** [comment]
Got it. Thanks for the solution and the explanation. And that's an interesting observation about the relationship between `contextmenu` and `longpress`.

I noticed that after adding `onLongPress={() => {}}` I'm now getting a short haptic feedback event when the `onLongPress` event fires (on Chrome at least). It's similar to the haptic feedback that occurs when a long press is used to open a context menu. Do you know if it's possible to disable this haptic feedback? Disabling it outside of React has been discussed a bit [here](https://stackoverflow.com/questions/46087752/how-to-completely-remove-mobile-browser-long-press-vibration), but I wasn't sure how to translate their solution to work with Pressable.

**Comment by necolas** [comment]
You should be able to set both those CSS properties on your pressable element.

**Comment by elliotwaite** [comment]
I've tried setting the CSS properties that were mentioned in that StackOverflow discussion, but they didn't seem to remove the haptic feedback ([demo](https://codesandbox.io/s/pressable-press-and-hold-test-gdpc5e?file=/src/global.css)).

It was more the `e.preventDefault()` and `e.stopPropagation()` for the `touchstart` and `touchend` events (as mentioned in [this answer](https://stackoverflow.com/a/70192263/2577315)) that I didn't know how to translate over to using with Pressable. I tried adding `e.preventDefault()` and `e.stopPropagation()` to `onPress`, `onPressIn`, `onPressOut`, and `onLongPress` ([demo](https://codesandbox.io/s/pressable-press-and-hold-test-gdpc5e?file=/src/App.js)), but those changes didn't seem to remove the haptic feedback either.

Do you have any other suggestions for things I could try?

**Comment by necolas** [comment]
No other suggestions. Until now I wasn't aware that browsers didn't cancel the haptic feedback along with the rest of the default behaviour.

The pressable callbacks are built on the responder system abstraction, which depends on touch events but doesn't alter their propagation or expose them for further modification. Having said that, it sounds like this might be an issue that we could/should fix somewhere in this chain of abstractions rather than in your product code.

**Comment by elliotwaite** [comment]
Okay, got it. Since your suggestion of adding `onLongPress={() => {}}` resolved the original issue, I opened a new issue that focuses on the haptic feedback issue: https://github.com/necolas/react-native-web/issues/2352

Thanks again for helping me resolve the original issue.
